src/modules/user-auth/components/LoginForm/useLoginForm.ts

```diff
@@ -13,39 +12,19 @@ import { useLayoutModal } from '@modules/layouts/contexts/LayoutModalContext';
 export const useLoginForm = () => {
   const { t } = useTranslation();
   const { loginModal } = useLayoutModal();
+  const [submitError, setSubmitError] = useState('');
 
   const formSchema = LoginFormSchema(t);
   const form = useForm<LoginFormModel>({
     defaultValues: formSchema.getDefault(),
     resolver: yupResolver(formSchema),
   });
-  const { control } = form;
 
-  const [submitError, setSubmitError] = useState('');
   const onSubmit = form.handleSubmit(async (data) => {
-    await AuthService.signInWithEmail(
-      data.email,
-      data.password,
-      data.keepLogIn
-    ).catch((err) => setSubmitError(err.error_description || err.message));
-
-    loginModal.hide();
+    AuthService.signInWithEmail(data.email, data.password, data.keepLogIn)
+      .then(() => loginModal.hide())
```

Comment:
we can keep the `await` here.

Reply:
Nah, that is the source of the bug in the first place. I recommend not mixing `await` and `then/catch` anymore since it produces more complexity. Any logic can be performed after both `try` and `catch` can be done before that anyway since there's practically no difference now.

Reply:
oh right, after it throws, it still goes on to the next line.